server/auth/auth.controller.js

```diff
@@ -0,0 +1,247 @@
+  const { token } = req.params;
+  const user = await AuthModel.findOne({
+    verifyAccountToken: token
+  });
+  if (!user) return next(new ErrorResponse('User not found', 404));
+  user.verifyAccountToken = undefined;
+  user.isAccountVerified = true;
+  await user.save();
+  return res.status(200).json({ success: true });
+});
+
+exports.forgetPasswordSendToken = asyncHandler(async (req, res, next) => {
+  const { error } = forgetPasswordVal(req.body);
+  if (error) return next(new ErrorResponse(error.details[0].message, 400));
+  const resetToken = crypto.randomBytes(20).toString('hex');
+
+  const { email } = req.body;
+  const user = await AuthModel.findOne({ email });
+  if (!user) return next(new ErrorResponse('User not found', 404));
+
+  const token = crypto
+    .createHash('sha256')
+    .update(resetToken)
+    .digest('hex');
+  user.resetPasswordToken = token;
+  user.resetPasswordExpire = Date.now() + 10 * 60 * 1000;
+
+  await user.save();
+  await sendMailForgetPasswordToken(user.username, user.email, token);
+  return res.status(200).json({ success: true, token });
+});
+
+exports.forgetPasswordRecieveToken = asyncHandler(async (req, res, next) => {
+  const { token } = req.params;
+  const user = await AuthModel.findOne({
+    resetPasswordToken: token,
+    resetPasswordExpire: { $gt: Date.now() }
+  });
+  if (!user) return next(new ErrorResponse('Token Expired !!', 401));
+  await user.save();
+  return res.status(200).json({ success: true, token });
+});
+
+exports.resetPassword = asyncHandler(async (req, res, next) => {
+  const { error } = resetPasswordVal(req.body);
+  if (error) return next(new ErrorResponse(error.details[0].message, 400));
+
+  const user = await AuthModel.findOne({
+    resetPasswordToken: req.params.token,
+    resetPasswordExpire: { $gt: Date.now() }
+  });
+  user.resetPasswordToken = undefined;
+  user.resetPasswordExpire = undefined;
+  user.isAccountVerified = true;
+
+  user.password = await bcrypt.hash(req.body.password, 10);
+  await user.save();
+  return res.status(200).json({ success: true });
+});
+
+exports.createNewPassword = asyncHandler(async (req, res, next) => {
+  const { oldPassword, newPassword } = req.body;
+  const { error } = createNewPasswordVal(req.body);
+  if (error) return next(new ErrorResponse(error.details[0].message, 400));
+
+  const user = await AuthModel.findById(req.user._id).select('+password');
+
+  const isMatched = await bcrypt.compare(oldPassword, user.password);
+  if (isMatched) {
+    user.password = await bcrypt.hash(newPassword, 10);
+    await user.save();
+    return res.status(200).json({ success: true });
+  }
+  return next(new ErrorResponse('Old password is not matching', 401));
+});
+
+exports.updateDetails = asyncHandler(async (req, res, next) => {
+  let { username, email, phoneNumber } = req.body;
+  if (!username) username = req.user.username;
+  if (!email) email = req.user.email;
+  if (!phoneNumber) phoneNumber = req.user.phoneNumber;
```

Comment:
Explain this line. Did you take whole user info into `req.user`?

Reply:
from my code, everything except password